src/org/folio/rest_v2/Eureka.groovy

```diff
@@ -0,0 +1,95 @@
+package org.folio.rest_v2
+
+import org.folio.models.EurekaTenant
+import org.folio.utilities.RequestException
+
+class Eureka extends Authorization {
+
+  /**
+   * EurekaTenant object contains Master Tenant configuration for Eureka.
+   * @attribute 'tenantId'     Default: "master"
+   * @attribute 'clientId'     Default: "folio-backend-admin-client"
+   */
+  public EurekaTenant masterTenant
+
+  Eureka(Object context, String eurekaDomain, EurekaTenant masterTenant, boolean debug = false) {
+    super(context, eurekaDomain, debug)
+    this.masterTenant = masterTenant
+  }
+
+  void createTenant(EurekaTenant tenant) {
+    if (isTenantExist(tenant.tenantId)) {
+      logger.warning("Tenant ${tenant.tenantId} already exists!")
+      return
+    }
+
+    String url = generateUrl("/tenants")  // Tenant Manager URL
```

Comment:
in our case that will not work, we're enforcing static URL for it: http://mgr-tenants.
Not a public one.

Reply:
Maybe it should be like map with static values in EurekaTenant class which refer to such components